src/Components/DrillIllustration.js

```diff
@@ -260,14 +203,132 @@ const DrillIllustration = props => {
     );
   };
 
-  return (
-    <GestureRecognizer style={styles.container} onSwipeLeft={checkAnimationFast} config={swipeConfig}>
-      {checkSwitch()}
-    </GestureRecognizer>
-  );
+  const checkSwitch = () => {
+    if (activeIndex === props.drill.steps.length) {
+      return displayFinish();
+    } else if (!currentStep) {
+      return <View />; // bad state, but let's not crash
+    } else {
+      switch (props.drill.steps[activeIndex].illustrationType) {
+        case IllustrationType.ANIMATION:
+          return displayAnimation(props.drill.steps[activeIndex]);
+        case IllustrationType.YOUTUBE:
+          return displayYoutube(props.drill.steps[activeIndex]);
+        case IllustrationType.VIMEO:
+          return displayVimeo(props.drill.steps[activeIndex]);
+        default:
+          return <Text>No visual content for this drill</Text>;
+      }
+    }
+  };
+
+  const displayAnimation = ({ illustrationSource, title }, index) => {
+    const isFirstStep = isMultipleSteps ? index === 0 : true;
```

Comment:
I guess you don't need to check `isMultipleSteps` because `index === 0` is always the condition for being the first step?